### PR TITLE
PXP-476: [CLI] Develop the config synthesizer command

### PR DESCRIPTION
**Pull request by jk-gan**
## Summary

Ticket: [PXP-476](https://mindvalley.atlassian.net/browse/PXP-476)

### Added
- [x] add `wukong dev config-synthesizer` command
Screenshot 2023-03-30 at 4 48 37 PM

[PXP-476]: https://mindvalley.atlassian.net/browse/PXP-476?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/80#" title="Author self-approved">jk-gan</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: src/commands/dev/config_synthesizer.rs
Comment (jk-gan) [reply src/commands/dev/config_synthesizer.rs:19]: hmm wait, I didn't notice this `panic!` here. Let me change it

File: src/commands/dev/config_synthesizer.rs
Comment (jk-gan) [reply src/commands/dev/config_synthesizer.rs:32]: Good catch!

File: src/services/vault/client.rs
Comment (onimsha) [inline src/services/vault/client.rs:56]: for prod, the Vault `base_url`  also the same as the dev environment. So we can just use the same url here.

File: tests/dev.rs
Comment (onimsha) [inline tests/dev.rs:111]: I don't understand. Why all the comments in this file ?

File: src/services/vault/client.rs
Comment (jk-gan) [reply src/services/vault/client.rs:56]: This is useful for testing, so in non prod environment we can use `WUKONG_DEV_VAULT_API_URL` to inject the vault url, e.g. mock server url.

File: tests/dev.rs
Comment (jk-gan) [reply tests/dev.rs]: Initially it was because there are dynamic values in the output, so the snapshot will failed for each text run. I just found a way to solve this. Fixed.

**Comment by jk-gan** [comment]
Updated the PR so now the command is fault tolerance, the program wouldn't fault in the middle if error occur.
Screenshot 2023-03-30 at 2 16 55 PM

**Comment by onimsha** [comment]
@jk-gan I tested this PR today, however I got a strange situation when the command just return without any indicator of success or failure, just an empty return. I will need to hook up with you first thing tmr morning to go through this .

**Review by onimsha — APPROVED** [review]

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/80#" title="Author self-approved">jk-gan</a>*, *<a href="https://github.com/mindvalley/wukong-cli/pull/80#pullrequestreview-1370128895" title="Approved">onimsha</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
